modules/experimental-layers/src/gpu-grid-layer/gpu-grid-layer.js

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2015 - 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {CompositeLayer} from '@deck.gl/core';
+
+import GPUGridCellLayer from './gpu-grid-cell-layer';
+
+import {pointToDensityGridData} from './gpu-grid-utils';
+import GPUGridAggregator from '@deck.gl/experimental-layers/utils/gpu-grid-aggregator';
+
+const MINCOLOR = [0, 0, 0, 255];
+const CPU_MAXCOLOR = [255, 0, 0, 255];
+const GPU_MAXCOLOR = [0, 255, 0, 255];
+
+const defaultProps = {
+  // elevation
+  elevationScale: 1,
+
+  // grid
+  cellSize: {type: 'number', min: 0, max: 1000, value: 1000},
+  coverage: {type: 'number', min: 0, max: 1, value: 1},
+  getPosition: x => x.position,
+  extruded: false,
+  fp64: false,
+  pickable: false, // TODO: Enable picking with GPU Aggregation
+
+  // Optional settings for 'lighting' shader module
+  lightSettings: {},
+
+  // GPU Aggregation
+  gpuAggregation: true
+};
+
+export default class GPUGridLayer extends CompositeLayer {
+  initializeState() {
+    const {gl} = this.context;
+    const options = {
+      id: `${this.id}-gpu-aggregator`,
+      shaderCache: this.context.shaderCache
+    };
+    this.state = {
+      gpuGridAggregator: new GPUGridAggregator(gl, options)
+    };
+  }
+
+  updateState({oldProps, props, changeFlags}) {
+    const reprojectNeeded = this.needsReProjectPoints(oldProps, props);
+
+    if (changeFlags.dataChanged || reprojectNeeded) {
+      // project data into hexagons, and get sortedBins
+      this.getLayerData();
+    }
+  }
+
+  needsReProjectPoints(oldProps, props) {
+    return oldProps.cellSize !== props.cellSize || oldProps.gpuAggregation !== props.gpuAggregation;
```

Comment:
needs reproject also need to check whether `getPosition` updateTrigger

Reply:
Addressed, looks like this needed for GridLayer too #1953  ?